shared/desktop/app/app-state.js

```diff
@@ -115,45 +115,118 @@ export default class AppState {
   }
 
   checkOpenAtLogin() {
-    if (!this.state.openAtLogin) {
-      console.log('Skip setting login item due to user pref')
-      return
-    }
-    if (__DEV__) {
-      console.log('Skip setting login item due to dev env')
-      return
-    }
     console.log('Setting login item due to user pref')
+
     this.setOSLoginState()
   }
 
   setOSLoginState() {
-    // Comment this out if you want to test auto login stuff w/ a dev build, otherwise do a package build
     if (__DEV__) {
       console.log('Skipping auto login state change due to dev env. ')
       return
     }
+    // Comment this out if you want to test auto login stuff
+
     const isDarwin = process.platform === 'darwin'
-    // Electron has a bug where setting this to false fails!
+    const isWindows = process.platform === 'win32'
+    // Electron has a bug where app.setLoginItemSettings() to false fails!
     // https://github.com/electron/electron/issues/10880
     if (isDarwin) {
-      const applescript = require('applescript')
-      try {
-        const appName = __DEV__ ? 'Electron' : 'Keybase'
-        const command = this.state.openAtLogin
-          ? `tell application "System Events" to make login item at end with properties {path:"${appBundlePath() ||
-              ''}", hidden:false, name:"${appName}"}`
-          : `tell application "System Events" to delete login item "${appName}"`
-        applescript.execString(command, (err, result) => {
+      this.setDarwinLoginState()
+    } else if (isWindows) {
+      this.setWinLoginState()
+    }
+  }
+
+  setDarwinLoginState() {
+    const applescript = require('applescript')
+    this.checkMultiDarwinLoginItems()
```

Comment:
`checkMultiDarwinLoginItems` removes everything but the first entry...but isn't it possible we add another entry below, so we'll end up with two?

Reply:
No, I went to some trouble to first look and see if there is one, and not add it if present

Reply:
Ah, you're right!